synapse/rest/client/pusher.py

```diff
@@ -53,6 +54,11 @@ async def on_GET(self, request: SynapseRequest) -> Tuple[int, JsonDict]:
 
         filtered_pushers = [p.as_dict() for p in pushers]
 
+        for pusher in filtered_pushers:
+            if self._msc3881_enabled:
+                pusher["org.matrix.msc3881.enabled"] = bool(pusher["enabled"])
+            del pusher["enabled"]
+
```

Comment:
Not critical, but: the explicit cast to bool looks weird here. Could avoid it by looping over e.g. `payload, pusher in zip(filtered_pushers, pushers)`.

(I don't understand why the adjective `filtered` was chosen here. Before your change there was absolutely no filtering going on AFACIS?)

Reply:
Yeah that bit needs to be reworked - especially since I've realised it means pushers with NULL `enabled` columns will be marked as disabled, which is not the expected behaviour.

Reply:
Oh good spot. Sorry I missed this.

Reply:
I've moved the cast to bool to the database code since it happens because of the SQLite/PostgreSQL difference in storing booleans (basically we don't want to be showing `1` instead of `true`).

> (I don't understand why the adjective filtered was chosen here. Before your change there was absolutely no filtering going on AFACIS?)

Yeah I'm not entirely sure why it's not e.g. `pusher_dicts` or something like that. `filtered` make no sense to me too in this context. I've renamed it.

Reply:
Discussion in synapse-dev: the cast is because sqlite gives us an int because it doesn't have boolean columns. (So the annotation PusherConfig.enabled: bool wasn't upheld!

Brendan says he plans to make this cast at the point of creating a PusherConfig, which I'm very grateful for :+1: